packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -143,9 +168,18 @@ const helpers = {
           httpContent,
           Option.fold(
             () => {
-              logger.warn(`Unable to find a content for ${mediaTypes}`);
-              return Either.left<Error, IHttpNegotiationResult>(
-                ProblemJsonError.fromTemplate(NOT_ACCEPTABLE, `Unable to find content for ${mediaTypes}`),
+              return pipe(
+                whenNoContent(logger, response, headers || [], mediaTypes),
+                Option.fold(
+                  () => {
+                    logger.warn(warnMsg(mediaTypes));
+
+                    return Either.left<Error, IHttpNegotiationResult>(
+                      ProblemJsonError.fromTemplate(NOT_ACCEPTABLE, `Unable to find content for ${mediaTypes}`),
+                    );
+                  },
+                  r => Either.right(r),
```

Comment:
You do not need to use `fold` if all you need is convert this to an `Either`

`Either.fromOption` is the function you're looking for.